### PR TITLE
refactor to use hooks

### PR DESCRIPTION
**Pull request by kstulgys**
update react, react-dom, delivery client packages and refactor code to use hooks

### CODE REVIEW

**Comment by makma** [comment]
Hi @kstulgys, thank you for the pull request. The usage of hooks seems to be proper and valid here, the code looks good to me, however, I'm not familiar with the tutorial so the guys from @Kentico/customer-education should consider if the code aligns with it.

Moreover, I'm not sure about changed code formatting (missing semicolons, added whitespaces, omitting parentheses in arrow function) - this should CustEd team consider as well.

There are also some confusing limitations while using hooks at first (https://reactjs.org/docs/hooks-overview.html#rules-of-hooks) so I'd suggest considering using proper linter plugin.

**Comment by jancerman** [comment]
Hey @kstulgys, the refactored version looks good. I think the tutorial updates around the code will be minimal, which is great. Can you please fix the linting issues @makma mentioned earlier so that we can merge this and update the tutorial?

**Comment by kstulgys** [comment]
Hey guys, thanks for the feedback. I have changed the formating. I had some issues with my local branch being out of sync with this one so you might see these weird commits.

File: src/ArticleView.js [hunk 1]
```diff
@@ -1,100 +1,51 @@
...
-        </div>
-      );
-    }
-  }
+  };
+
+  useEffect(() => {
+    const subscription = fetchArticle(match.params.slug);
+    return () => subscription.unsubscribe();
+  }, []);
```
Comment (ChristopherJennings) [inline src/ArticleView.js]: Adding `match.params.slug` to the array on line 32 should prevent this effect from running on every render unless the slug changes.

**Review by ChristopherJennings — CHANGES_REQUESTED** [review]
Overall, looks good to me. Just the one minor change as noted on the dependencies array in useEffect on line 32 of `ArticleView.js

File: src/ArticleView.js (hunk 1 above)
Comment (petrsvihlik) [reply src/ArticleView.js]: @kstulgys could you please adjust the code? we'll be happy to merge it then :)

Comment (kstulgys) [reply src/ArticleView.js]: @petrsvihlik Sorry for late response, I hope this is still relevant, added in the useEffect hook dependency. As mentioned above :)

File: src/ArticleView.js [hunk 2]
```diff
@@ -1,100 +1,51 @@
...
+  }
+
+  return (
+    <div>
+      <Link to='/'>Home</Link>
+      <h1>{article.title.value}</h1>
+      <div
+        className='article_body'
+        dangerouslySetInnerHTML={{__html: article.body_copy.resolveHtml()}}
+        onClick={(event) => this.handleClick(event, article.body_copy)}
```
Comment (jancerman) [inline src/ArticleView.js]: This here fails with TypeError: Cannot read property 'handleClick' of undefined when you click inside article detail. For example, somewhere in the text.

**Review by jancerman — CHANGES_REQUESTED** [review]
Hey @kstulgys, I've briefly run and tested the app and ran into a TypeError. It occurs when you click within article detail. This doesn't occur on the master branch. Please check it and add validation.

File: src/ArticleView.js (hunk 2 above)
Comment (kstulgys) [reply src/ArticleView.js]: Should be fixed now :)

Comment (jancerman) [reply src/ArticleView.js]: Works as intended now! Thanks.

**Review by jancerman — APPROVED** [review]

**Comment by jancerman** [comment]
Looks like everything is as it should be. Thanks again, @kstulgys. I'll merge & close this PR.

**Pull request merged**